src/frontend/src/handler/declare_cursor.rs

```diff
@@ -166,7 +166,7 @@ pub fn create_batch_plan_for_cursor(
     let out_fields = FixedBitSet::from_iter(0..batch_log_seq_scan.core().schema().len());
     let out_names = batch_log_seq_scan.core().column_names();
     // Here we just need a plan_root to call the method, only out_fields and out_names will be used
-    let mut plan_root = PlanRoot::new(
+    let plan_root = PlanRoot::new_with_batch_plan(
```

Comment:
Elaborate more the motivation of this PR since https://github.com/risingwavelabs/risingwave/pull/16487 got merged. `Cursor` will generate a batch from scratch, so we add a plan phase here to make the `PlanRoot` more robust.